### DISCUSSION TITLE
Direct Paths

### DISCUSSION THREAD

**Discussion by Bartyboy** [croneter/PlexKodiConnect#1527] [Q&A] [answered]
Hi

Thanks for your great Plugin.
I've trouble in configurating Direct Paths in the Plugin.
I tested several hours and read your explanations carefully and many times.
But I'm failing with all my tests.

My configuration ist the following:
I've a Plexserver on a Synology NAS.
This are my Plex 

This are the nfs shares:
nfs://192.168.0.50/volume1/Movies
nfs://192.168.0.50/volume1/Kids movies
nfs://192.168.0.50/volume1/Series
nfs://192.168.0.50/volume1/Kids series
...

The Plex-librarys are:
/volume1/Movies
/volume1/Kids movies
/volume1/Series
/volume1/kids series
...

I'm not shure, If I should set-up option 1 or option 2.
I think it should be option 1, but I don't know how to share ...Movies and Kids movies as par example "nfs://myNAS/plex-movies/Movies". On my synology NAS I can't configure something on "volume1". The shares with nfs and smb rights are "Movies", "Kids Movies".

But if I test to replace following:
plex path: /volume1/Movies
with: nfs://192.168.0.50/volume1/Movies or nfs://192.168.0.50/volume1
Plexkodiconnect say's, that the files are no longer available.

If I should configure the option 2, I'm stucking on point 5:
"In our example, simply mount smb://myNas/ directly to e.g. /media/myNas and you're done"
How can I mount this in Plex? Or should I configure something on the nas first?

I hope you can help me, I would like the plugin in direct mode. With the plug-in mode it works, but on a firestick with only wlan is direct link the best thing.

**Comment by croneter** [comment]
Above all else, make sure that you can use [Kodi's file manager](https://kodi.wiki/view/File_manager) to browse to your Plex media. If that does not work, direct paths won't work. 

> But if I test to replace following:
> plex path: /volume1/Movies
> with: nfs://192.168.0.50/volume1/Movies or nfs://192.168.0.50/volume1
> Plexkodiconnect say's, that the files are no longer available.

The "with" seems wrong. It'll be something like this, depending how you make your `Movies` folder available:
```
nfs://192.168.0.50/Movies
```
What path do you see using other clients than Kodi/PKC when browsing to that network location? That's what you need to plug-in here.

**Reply by Bartyboy** [reply]
Thanks for your answer.
I can browse with kod's file manager to my Plex media and can start it directly from kodi. The file manager shows "nfs://192.168.0.50/volume1/Movies/Movie/movie.mkv" if I play the file.
I can't browse through it if I want use smb, but I don't know why. It say's, there ist a problem with the credentials, but they are correct in the password.xml file. Other clients can connect through smb without problems. On smb the share is definitly "smb://192.168.0.50/Movies" but I don't know how to test nfs on windows or android. But Synology Share manager shows me the following mount path: /volume1/movies.
Should it work with my different movie folders or should I follow the substitute path 2?

**Reply by croneter** [reply]
Issue here is smb itself, there was a breaking change on the Kodi 19 Matrix side of things. See e.g. https://forum.kodi.tv/showthread.php?tid=360606

**Comment by Bartyboy** [comment, marked as answer]
I found the problem now.
If I activate "Escape special characters in the path" (sorry, I don't know the real name, it's on german in my case), it put's a ' in front of the path and kodi means it's an unsupported protocol (/smb)
The path in the log is like this:
/smb://192.168.0.50/Movies
If "Escape special characters in the path" is disabled, I can play the files directly. Now, I have to reorganize my media because it only finds the media in the "smb://192.168.0.50/Movies" and "smb://192.168.0.50/Series" folder. "smb://192.168.0.50/Kids Movies" can't be found.
I will create a shared folder "smb://192.168.0.50/media" and will transfer all subfolders in this shared folder. I think, this should work.

**Reply by croneter** [reply]
Out thanks for that! Made a new issue: #1529

**Reply by ricksrod** [reply]
Hi @Bartyboy - do you mind sharing a screenshot of your paths are replaced in PKC? Is it:

volume1/Movies
replace with:
smb://192.168.0.50/Movies

?
